pkg/feature/api/feature.go

```diff
@@ -2390,6 +2391,222 @@ func (s *FeatureService) EvaluateFeatures(
+		Evaluations:        featureEvaluations,
+		ArchivedFeatureIds: archivedFS,
+	}, nil
+}
+
+func (s *FeatureService) evaluateLatestFeatures(
+	ctx context.Context,
+	req *featureproto.DebugEvaluateFeaturesRequest,
+	localizer locale.Localizer,
+) ([]*featureproto.Evaluation, []string, error) {
+	fs, err, _ := s.flightgroup.Do(
+		req.EnvironmentId,
+		func() (interface{}, error) {
+			return s.getFeatures(ctx, req.EnvironmentId)
+		},
+	)
+	if err != nil {
+		s.logger.Error(
+			"Failed to list features",
+			log.FieldsFromImcomingContext(ctx).AddFields(
+				zap.Error(err),
+				zap.String("environmentId", req.EnvironmentId),
+			)...,
+		)
+		dt, err := statusInternal.WithDetails(&errdetails.LocalizedMessage{
+			Locale:  localizer.GetLocale(),
+			Message: localizer.MustLocalize(locale.InternalServerError),
+		})
+		if err != nil {
+			return nil, nil, statusInternal.Err()
+		}
+		return nil, nil, dt.Err()
+	}
+	latestFeatures := fs.([]*featureproto.Feature)
+	latestEvaluationResult := make([]*featureproto.Evaluation, 0)
+	archivedFS := make([]string, 0)
+	for i := range req.Users {
+		userEvaluations, err := s.evaluateFeatures(
+			ctx,
+			latestFeatures, req.Users[i], req.EnvironmentId, req.Tag, localizer)
+		if err != nil {
+			s.logger.Error(
+				"Failed to evaluate features",
+				log.FieldsFromImcomingContext(ctx).AddFields(
+					zap.Error(err),
+					zap.String("environmentId", req.EnvironmentId),
+				)...,
+			)
+			dt, err := statusInternal.WithDetails(&errdetails.LocalizedMessage{
+				Locale:  localizer.GetLocale(),
+				Message: localizer.MustLocalize(locale.InternalServerError),
+			})
+			if err != nil {
+				return nil, nil, statusInternal.Err()
+			}
+			return nil, nil, dt.Err()
+		}
+
+		for _, f := range req.Features {
+			for _, e := range userEvaluations.Evaluations {
+				if f.Id == e.FeatureId && (f.FeatureVersion == e.FeatureVersion || f.FeatureVersion == 0) {
+					latestEvaluationResult = append(latestEvaluationResult, e)
+					break
+				}
+			}
+		}
+		archivedFS = append(archivedFS, userEvaluations.ArchivedFeatureIds...)
+	}
+	return latestEvaluationResult, archivedFS, nil
+}
+
+func (s *FeatureService) evaluateOlderFeatures(
+	ctx context.Context,
+	req *featureproto.DebugEvaluateFeaturesRequest,
+	localizer locale.Localizer,
+) ([]*featureproto.Evaluation, []string, error) {
+	olderFeatures, err := s.getOlderVersionFeatures(ctx, req.Features, req.EnvironmentId)
+	if err != nil {
+		return nil, nil, err
+	}
+	for i := range olderFeatures {
+		// to avoid duplicate feature IDs when doing topology sort, currently adding version to the ID
+		// TODO: currently not able to evaluate depend features in the past
+		olderFeatures[i].Id = fmt.Sprintf("%s:%d", olderFeatures[i].Id, olderFeatures[i].Version)
+	}
+
+	evaluations := make([]*featureproto.Evaluation, 0)
+	archivedFS := make([]string, 0)
+
+	for i := range req.Users {
+		userEvaluations, err := s.evaluateFeatures(
+			ctx,
+			olderFeatures,
```

Comment:
The old features here don't include the flags being used in the prerequisites and rules targeting.